### PR TITLE
Add sort operator support in velox

### PR DESCRIPTION
**Pull request by JkSelf**

### CODE REVIEW

**Comment by JkSelf** [comment]
@rui-mo Do you have any comment?

File: velox/substrait/SubstraitToVeloxPlan.cpp [hunk 1]
```diff
@@ -348,6 +348,58 @@ core::PlanNodePtr SubstraitVeloxPlanConverter::toVeloxPlan(
...
+
+  std::vector<core::FieldAccessTypedExprPtr> sortingKeys;
+  std::vector<core::SortOrder> sortingOrders;
+
+  const auto& sorts = sortRel.sorts();
+  sortingKeys.reserve(sorts.size());
+  sortingOrders.reserve(sorts.size());
+
+  for (const auto& sort : sorts) {
+    if (sort.sort_kind_case() == 2) {
```
Comment (rui-mo) [inline velox/substrait/SubstraitToVeloxPlan.cpp:370]: What is the for?

File: velox/substrait/SubstraitToVeloxPlan.cpp [hunk 2]
```diff
@@ -348,6 +348,58 @@ core::PlanNodePtr SubstraitVeloxPlanConverter::toVeloxPlan(
...
+          sortingOrders.emplace_back(core::kDescNullsLast);
+          break;
+        default:
+          VELOX_FAIL("Sort direction is not support in SortRel");
+      }
+    }
+
+    if (sort.has_expr()) {
+      sortingKeys.emplace_back(
+          std::dynamic_pointer_cast<const core::FieldAccessTypedExpr>(
```
Comment (rui-mo) [inline velox/substrait/SubstraitToVeloxPlan.cpp:394]: What if `sort.expr()` is not FieldAccessTypedExpr? Should error be thrown?

File: velox/substrait/SubstraitToVeloxPlanValidator.cpp [hunk 3]
```diff
@@ -65,6 +65,63 @@ bool SubstraitToVeloxPlanValidator::validateInputTypes(
...
+            SortField_SortDirection_SORT_DIRECTION_DESC_NULLS_LAST:
+          break;
+        default:
+          return false;
+      }
+    }
+
+    if (sort.has_expr()) {
+      try {
+        auto expression = exprConverter_->toVeloxExpr(sort.expr(), rowType);
```
Comment (rui-mo) [inline velox/substrait/SubstraitToVeloxPlanValidator.cpp:112]: How about also validate whether the `sort.expr()` is field?

File: velox/substrait/SubstraitToVeloxPlan.cpp [hunk 4]
```diff
@@ -348,6 +348,60 @@ core::PlanNodePtr SubstraitVeloxPlanConverter::toVeloxPlan(
...
+        break;
+      default:
+        VELOX_FAIL("Sort direction is not support in SortRel");
+    }
+
+    if (sort.has_expr()) {
+      auto expression = exprConverter_->toVeloxExpr(sort.expr(), inputType);
+      auto expr_field =
+          dynamic_cast<const core::FieldAccessTypedExpr*>(expression.get());
+      VELOX_CHECK(
```
Comment (JkSelf) [inline velox/substrait/SubstraitToVeloxPlan.cpp:391]: Add check whether the sort.expr is FieldAccessTypesExpr.

File: velox/substrait/SubstraitToVeloxPlan.cpp (hunk 1 above)
Comment (JkSelf) [reply velox/substrait/SubstraitToVeloxPlan.cpp]: Remove this check. Because this check has done in `sort.direction()` method.

**Review by rui-mo — APPROVED** [review]
LGTM.

**Pull request merged**